### ISSUE TITLE
Weird query evaluation behaviour

### ISSUE THREAD

**Issue by psiotwo** [ontodev/robot#1112]
Given the following input:

```
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix obo: <http://purl.obolibrary.org/obo/> .
@prefix oboInOwl: <http://www.geneontology.org/formats/oboInOwl#> .

oboInOwl:hasDbXref a owl:AnnotationProperty .

obo:PR_Q9NTK1 a owl:Class .

[
    a owl:Axiom ;
    owl:annotatedSource obo:PR_Q9NTK1 ;
    owl:annotatedProperty oboInOwl:hasRelatedSynonym ;
    owl:annotatedTarget "C10orf10" ;
    oboInOwl:hasDbXref "UniProtKB:Q9NTK1" ;
] .
```

the presence of `oboInOwl:hasDbXref a owl:AnnotationProperty .` triple does affect the result of a simple query:

```
PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>

DELETE {
    ?concept oboInOwl:hasRelatedSynonym ?synonym .
} WHERE {
    ?concept oboInOwl:hasRelatedSynonym ?synonym
}
```

With this triple in, the query  `robot query -i input.ttl -u query.rq -o output.ttl` evaluation materializes the reified axiom as :

```
<http://purl.obolibrary.org/obo/PR_Q9NTK1> rdf:type owl:Class ;
                                           <http://www.geneontology.org/formats/oboInOwl#hasRelatedSynonym> "C10orf10" .
```
Without that triple, the query removes the triple.

I consider this behaviour inconsistent. Would expect both queries to behave the same.

**Comment by jamesaoverton** [comment]
In order to support various OWL formats and imports, ROBOT loads all input data through OWLAPI, and then sends it to Apache Jena if SPARQL operations are required.

My understanding is that your OWL Annotation Axiom implies that there's a triple to annotate, and your example is not well-formed as you presented it. So I guess that OWLAPI is adding the triple when `input.ttl` is loaded.

**Comment by jamesaoverton** [comment]
Maybe I didn't quite understand. OWL Axiom Annotations require the predicate to be declared as an OWL Annotation Property, which is a common source of confusion that you can see elsewhere on this tracker.

**Comment by psiotwo** [comment]
Yes, I am aware of this - this seems just another example of a mismatch of the RDF-entailment over OWL. The query behaves
1) intuitively for invalid OWL
2) counterintuitively for valid OWL

**Comment by jamesaoverton** [comment]
SPARQL UPDATE is an RDF operation, not an OWL operation. How do you suggest that ROBOT should handle it?

**Comment by psiotwo** [comment]
OWL entailment regime might help (e.g. for handling the reified axioms) - https://www.w3.org/TR/sparql11-entailment/#OWLRDFBSEntRegime.

**Comment by jamesaoverton** [comment]
What tool would we use to implement that?

**Comment by psiotwo** [comment]
At some point I was working on https://github.com/kbss-cvut/owl2query - it is pretty dormant at present, but will try to look around and propose something here.
